### DISCUSSION TITLE
ttx -t – why are table tags case-sensitive?

### DISCUSSION THREAD

**Discussion by frankrolf** [fonttools/fonttools#3108] [General]
When using ttx on the command line, this works:
```ttx -t OS/2 font.otf```

but this doesn’t:
```ttx -t os/2 font.otf```

Of course, eventually I end up remembering which table tab is all-caps, and which isn’t – but I can see others making the same mistake. For example – why is it `avar`, `fvar`, but `MVAR`? My improvement suggestion would be to allow both upper-and lowercase spelling for all table tags.

Thanks!

**Comment by justvanrossum** [comment]
Table tags are case sensitive in the font, so it makes sense for them to be case sensitive in a command line interface.

While in the official set of tables there don't seem to be two table tags that would be same when case-folded, this isn't guaranteed for the future, and could also collide with any private tables.

**Reply by justvanrossum** [reply]
Generally: all lowercase tags come from Apple, all uppercase tags come from Microsoft, anything else must be mixed case. [citation needed]

**Reply by anthrotype** [reply]
ditto what Just said. For reference, the [TrueType spec](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6.html) mandates

> "Tag names consisting of all lower case letters are reserved for Apple's use."

**Comment by khaledhosny** [comment]
In addition to what @justvanrossum and @anthrotype said, there is at least one existing case of two font tables with the same tag but different case: Apple’s AAT has `feat` table and SIL’s Graphite has `Feat` table.

**Comment by frankrolf** [comment]
Thanks, learning something new every day!

Nevertheless, I find the behavior somewhat cumbersome. I understand that a workaround would require checking for these edge cases, so I don’t expect this suggestion to get much traction.

FWIW, a related issue: https://github.com/fonttools/fonttools/issues/972

**Reply by justvanrossum** [reply]
-1 from me for changing the current behavior. It's easy enough to see what the actual table tags are by running

    ttx -l path/to/font.otf

Regarding the related issue: a PR would be more than welcome.
